course02/homework00/joost/vector.h

```diff
@@ -0,0 +1,47 @@
+#pragma once
+#include <array>
+#include <stdexcept>
+
+class VectorInt
+{
+	public:
+		VectorInt():
+			mSize(0)
+		{}
+		
+		int size() const
+		{
+			return mSize;
+		}
+
+		int capacity() const
+		{
+			return mData.size();
+		}
+
+		void push_back(int value)
+		{
+			if(mSize == mData.size()){
+				throw std::out_of_range("vector at capacity");
+			}
+			mData[mSize++] = value;
+		}
+
+		int back(){
```

Comment:
This should be marked as `const`

Reply:
Also the corresponding overload is missing (the one that returns a ref, that is non const)